quinn-proto/src/token.rs

```diff
@@ -68,6 +79,22 @@ impl RetryToken {
             issued,
         })
     }
+
+    /// Ensure that this token validates an `Incoming`, and construct its token state
+    pub(crate) fn validate(
+        &self,
+        header: &InitialHeader,
+        server_config: &ServerConfig,
+    ) -> Result<IncomingTokenState, ValidationError> {
+        if self.issued + server_config.retry_token_lifetime > SystemTime::now() {
```

Comment:
Suggest inverting the condition for an early return in the error case, reducing rightward drift for the `Ok` case.